### PR TITLE
[CS-4300] Adds unit tests to profile slug validation screen

### PR DESCRIPTION
**Pull request by oidoug**
### Description

Adds unit tests to `useProfileSlugScreen.ts` with good coverage of edge cases, including api erroring out.

- [x] Completes #(CS-4300)

No visual changes.

### CODE REVIEW

File: cardstack/src/screens/Profile/ProfileSlugScreen/__tests__/useProfileSlugScreen.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}));
+
+describe('useProfileSlugScreen', () => {
+  const mockLazyValidateProfileSlugQuery = (
+    expectedResponse?: Partial<BusinessIDUniquenessResponse>,
+    success?: any
+  ) => {
+    (useLazyValidateProfileSlugQuery as jest.Mock).mockImplementation(() => [
+      jest.fn(),
+      { data: expectedResponse, error: success },
```
Comment (paulinhapenedo) [inline cardstack/src/screens/Profile/ProfileSlugScreen/__tests__/useProfileSlugScreen.test.ts]: nit: having a param named `success` being used in the error return is a bit odd.

**Review by paulinhapenedo — APPROVED** [review]
LGTM

File: cardstack/src/screens/Profile/ProfileSlugScreen/__tests__/useProfileSlugScreen.test.ts (hunk 1 above)
Comment (oidoug) [reply cardstack/src/screens/Profile/ProfileSlugScreen/__tests__/useProfileSlugScreen.test.ts]: Ah yeah, good catch.
